procurement_plan/models/procurement.py

```diff
@@ -0,0 +1,160 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+# For copyright and license notices, see __openerp__.py file in root directory
+##############################################################################
+from openerp import models, fields, api
+
+
+class ProcurementOrder(models.Model):
+    _inherit = 'procurement.order'
+
+    plan = fields.Many2one('procurement.plan', string='Plan')
+
+    @api.model
+    def _procure_orderpoint_confirm(self, use_new_cursor=False,
+                                    company_id=False):
+        procurement_obj = self.env['procurement.order']
+        move_obj = self.env['stock.move']
+        my_cursor = False
+        untreated_procurements = {}
+        untreated_moves = {}
+        if 'procurement_ids' not in self.env.context:
+            my_cursor = use_new_cursor
+        else:
+            cond = [('id', 'not in', self.env.context['procurement_ids'].ids),
+                    ('state', 'not in', ('cancel', 'done'))]
+            procurements = procurement_obj.search(cond)
+            for procurement in procurements:
+                my_vals = {'id': procurement.id,
+                           'state': procurement.state,
+                           }
+                untreated_procurements[(procurement.id)] = my_vals
+                cond = [('procurement_id', '=', procurement.id)]
+                moves = move_obj.search(cond)
+                if moves:
+                    for move in moves:
+                        my_vals = {'id': move.id,
+                                   'state': move.state,
+                                   }
+                        untreated_moves[(move.id)] = my_vals
+                    self.env.cr.execute('update stock_move set state=%s'
+                                        ' where id in %s',
+                                        ('cancel', tuple(moves.ids)))
+            if procurements:
+                self.env.cr.execute('update procurement_order set state=%s'
+                                    ' where id in %s',
+                                    ('cancel', tuple(procurements.ids)))
+        result = super(ProcurementOrder, self)._procure_orderpoint_confirm(
+            use_new_cursor=my_cursor, company_id=company_id)
+        if untreated_procurements:
+            for data in untreated_procurements:
```

Comment:
Aquí haces `for procurement_id, state in untreatead_procurements.iteritems():` y ya recuperas ambos valores.